### PR TITLE
Register all of the recently merged flowsheet UI modules

### LINKED ISSUE #781: fsapi errors while testing recently added flowsheet UI modules
These look like some extra handling is needed to account for more types of Pyomo objects, but superficially don't seem too hard to address.

image
image

### PR DESCRIPTION
**Pull request by lbianchi-lbl**
## Fixes/Resolves:

Part of #777.

## Summary/Motivation:

- All UI flowsheet modules need to be registered as an entry point in `setup.py` to be tested and available in the UI.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -149,6 +149,14 @@
...
+            "ENR = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.electrochemical_nutrient_removal.electrochemical_nutrient_removal_ui",
+            "CANDO_P = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.amo_1595_photothermal_membrane_candoP.amo_1595_ui",
+            "supercritical_sludge_to_gas = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.supercritical_sludge_to_gas.supercritical_sludge_to_gas_ui",
+            "PAA = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.peracetic_acid_disinfection.peracetic_acid_disinfection_ui",
+            "AMO 1690 = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.amo_1690.amo_1690_ui",
+            "HRCS = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.amo_1575_hrcs.hrcs_ui",
+            "groundwater_treatment = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.groundwater_treatment.groundwater_treatment_ui",
+            "dye_desalination = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.dye_desalination.dye_desalination_ui",
```
Comment (adam-a-a) [inline setup.py:159]: I might be losing my marbles, but are we missing one? I remember counting 13 and there are 12 here.

Comment (adam-a-a) [inline setup.py:159]:
```suggestion
            "dye_desalination = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.dye_desalination.dye_desalination_ui",
            "swine_wwt = watertap.examples.flowsheets.case_studies.wastewater_resource_recovery.swine_wwt.swine_wwt_ui",
```

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1575_hrcs/hrcs.py [hunk 2]
```diff
@@ -207,6 +207,7 @@ def add_costing(m):
     )
 
     m.fs.costing = ZeroOrderCosting(case_study_definition=source_file)
+    m.fs.costing.base_currency = pyunits.USD_2018
     m.fs.watertap_costing = WaterTAPCosting()
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1575_hrcs/hrcs.py]:
```suggestion
    m.fs.watertap_costing = WaterTAPCosting()
    m.fs.watertap_costing.base_currency = pyunits.USD_2020

```

File: setup.py (hunk 1 above)
Comment (lbianchi-lbl) [reply setup.py:159]: Thanks for spotting this, I've added this in a local commit.

File: watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1575_hrcs/hrcs.py (hunk 2 above)
Comment (lbianchi-lbl) [reply watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/amo_1575_hrcs/hrcs.py]: I know this is the opposite of what we discussed at the time, but setting the base currency on the `watertap_costing` side caused another (non-UI) test to fail, so I just blindly tried to switch the two (15709e45c8097f78d8aabed0893e88c586d2d9d6). If that's not appropriate, I'll try to see if I can reproduce the error to give more accurate information.

**Review by bknueven — APPROVED** [review]

**Review by TimBartholomew — APPROVED** [review]

**Pull request merged**